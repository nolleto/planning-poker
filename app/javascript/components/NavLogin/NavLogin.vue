<template>
  <div>NavLogin</div>
</template>

<script>
export default {
  name: 'NavLogin'
}
</script>

<style>

</style>
