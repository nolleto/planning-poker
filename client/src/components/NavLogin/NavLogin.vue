<template>
  <div>
    <div v-if="isUserSignedIn">
      <span>
        Hello
        <b>{{ username }}</b>!
      </span>

      <a @click="$emit('sign-out')">Sign out</a>
    </div>

    <div v-else>
      <a @click="$emit('sign-in')">Sign In</a>
      <a @click="$emit('sign-up')">Sign Up</a>
    </div>
  </div>
</template>

<script>
export default {
  name: 'NavLogin',

  props: {
    isUserSignedIn: {
      type: Boolean,
      required: true
    },

    username: {
      type: String
    }
  }
}
</script>

<style>

</style>
