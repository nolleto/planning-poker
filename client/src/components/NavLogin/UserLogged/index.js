export { default } from './UserLogged.vue'
