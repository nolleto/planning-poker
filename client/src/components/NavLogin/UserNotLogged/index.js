export { default } from './UserNotLogged.vue'
