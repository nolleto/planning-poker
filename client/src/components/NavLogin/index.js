export { default } from './NavLogin.vue'
