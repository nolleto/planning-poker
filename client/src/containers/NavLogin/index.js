import { connect } from 'vuex-connect'
import router from '../../routers'
import NavLogin from '../../components/NavLogin'

export default connect({
  gettersToProps: {
    isUserSignedIn: 'isUserSignedIn',
    username: 'username'
  },

  actionsToEvents: {
    'sign-out': dispatch => dispatch('signOut'),
    'sign-in': () => router.push('/users/sign_in'),
    'sign-up': () => router.push('/users/sign_up')
  }
})('NavLoginContainer', NavLogin)
