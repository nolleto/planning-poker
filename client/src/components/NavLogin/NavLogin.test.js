import NavLogin from './index'
import { shallowMount } from '@vue/test-utils'

const defaultProps = {
  isUserSignedIn: false
}

const shallowComponent = (opts = { propsData: defaultProps }) => {
  return shallowMount(NavLogin, opts)
}

describe('NavLogin component', () => {
  let wrapper

  describe('Snapshots', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    describe('When user is signed', () => {
      beforeAll(() => {
        wrapper.setProps({
          isUserSignedIn: true,
          username: 'Lucas Nascimento'
        })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When user is not signed', () => {
      beforeAll(() => {
        wrapper.setProps({ isUserSignedIn: false })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })

  describe('Event Listeners', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    describe('When user is signed', () => {
      beforeAll(() => {
        wrapper.setProps({ isUserSignedIn: true })
      })

      describe('When clicked in "Sign out" link', () => {
        beforeEach(() => {
          jest.spyOn(wrapper.vm, '$emit')
          wrapper.find('a').trigger('click')
        })

        it('emits "sign-out"', () => {
          expect(wrapper.vm.$emit).toHaveBeenCalledWith('sign-out')
        })
      })
    })

    describe('When user is not signed', () => {
      beforeAll(() => {
        wrapper.setProps({ isUserSignedIn: false })
      })

      describe('When clicked in "Sign In" link', () => {
        beforeEach(() => {
          jest.spyOn(wrapper.vm, '$emit')
          wrapper.find('a').trigger('click')
        })

        it('emits "sign-in"', () => {
          expect(wrapper.vm.$emit).toHaveBeenCalledWith('sign-in')
        })
      })

      describe('When clicked in "Sign Up" link', () => {
        beforeEach(() => {
          jest.spyOn(wrapper.vm, '$emit')
          wrapper.findAll('a').at(1).trigger('click')
        })

        it('emits "sign-up"', () => {
          expect(wrapper.vm.$emit).toHaveBeenCalledWith('sign-up')
        })
      })
    })
  })
})
