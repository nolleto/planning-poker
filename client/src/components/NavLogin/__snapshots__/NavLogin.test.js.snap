// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLogin component Snapshots When user is not signed matches 1`] = `
<div>
  <div>
    <a>
      Sign In
    </a>
     
    <a>
      Sign Up
    </a>
  </div>
</div>
`;

exports[`NavLogin component Snapshots When user is signed matches 1`] = `
<div>
  <div>
    <span>
      
      Hello
      
      <b>
        Lucas Nascimento
      </b>
      !
    
    </span>
     
    <a>
      Sign out
    </a>
  </div>
</div>
`;
